File: source/back/acnacional-core/src/main/java/com/certificadoranacional/ac/core/service/GrupoProdutoService.java
package com.certificadoranacional.ac.core.service;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.certificadoranacional.ac.core.model.GrupoProdutoRepresentation;

public interface GrupoProdutoService {

  GrupoProdutoRepresentation get(String id);

  GrupoProdutoRepresentation save(GrupoProdutoRepresentation rep);

  GrupoProdutoRepresentation update(GrupoProdutoRepresentation rep);

  Boolean delete(String id);

  Page<GrupoProdutoRepresentation> listByActive(String tipo, Pageable pageable);

  Page<GrupoProdutoRepresentation> list(String tipo, Pageable pageable);

}

File: source/front/acnacional-ng2/src/app/shared/services/grupo-produto.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';

import { Observable } from 'rxjs';

import { EnvironmentService } from './environment.service';
import { Logger } from './logger';

import { GrupoProduto, Page } from '../models';

@Injectable()
export class GrupoProdutoService {

  private url;

  constructor(private environmentService: EnvironmentService, private httpClient: HttpClient, private logger: Logger) {
    this.url = this.environmentService.getResourcesUrl() + '/grupo-produto';
    this.logger.debug(`URL:  ${this.url}`);
  }

  public get(id: string): Observable<GrupoProduto> {
    this.logger.debug(`GrupoProdutoService.get(${id});`);
    const getUrl = `${this.url}/${id}`;

    return this.httpClient.get<GrupoProduto>(getUrl);
  }

  public list(tipo: string, valid: boolean, text: string, page: number, size: number = 10): Observable<Page<GrupoProduto>> {
    this.logger.debug(`GrupoProdutoService.list();`);
    const type = valid ? 'active' : 'all';
    const listUrl = `${this.url}/?q=${text}&p=${page}&s=${size}&type=${type}&tipo_id=${tipo}`;

    return this.httpClient.get(listUrl);
  }

  public save(grupo: GrupoProduto): Observable<GrupoProduto> {
    this.logger.debug(`GrupoProdutoService.save(${grupo};`);
    const postUrl = `${this.url}`;

    return this.httpClient.post<GrupoProduto>(postUrl, grupo);
  }

  public update(grupo: GrupoProduto): Observable<GrupoProduto> {
    this.logger.debug(`GrupoProdutoService.update(${grupo};`);
    const postUrl = `${this.url}/${grupo.id}`;

    return this.httpClient.put<GrupoProduto>(postUrl, grupo);
  }

}
